pkg/tfbridge/info.go

```diff
@@ -363,6 +363,17 @@ type ResourceInfo struct {
 	// are accessed by other provider functions or by terraform. In particular, it can
 	// be used to perform upgrades on old pulumi state.  Should be used sparingly.
 	TransformFromState PropertyTransform
+
+	// Customizes inferring resource identity from state.
+	//
+	// The vast majority of resources define an "id" field that is recognized as the resource
+	// identity. This is the default behavior when ComputeID is nil. There are some
+	// exceptions, however, such as RandomBytes resource, that base identity on a different
+	// field ("base64" in the case of RandomBytes). ComputeID customization option supports
+	// such resources and will be called during Create and Read provider methods.
+	//
+	// This option is currently only supported for Plugin Framework based resources.
+	ComputeID func(state resource.PropertyMap) (resource.ID, error)
```

Comment:
Let's make sure that we can extend this method, and that logging and cancelation is available.

For example, if we want to make the ID depend on a field that the user has set as secret, we should probably warn (since ID cannot be secret).
```suggestion
	ComputeID func(ctx context.Contex, state resource.PropertyMap) (resource.ID, error)
```

Reply:
Yes, Context propagation is a good idea.